File: 기능구현/program_예상대진표.java
import java.util.*;
import java.io.*;


public class Solution {

    public int game(int k, double a, double b, int depth){

        if(depth == k){
            return depth;
        }

        if(a%2 == 0 && a-b==1){
            return depth;
        }
        else if(a%2==1 && b-a==1){
            return depth;
        }
        else{
            return game(k,Math.ceil(a/2),Math.ceil(b/2),depth+1);
        }
    }
    public int solution(int n, int a, int b)
    {
        int k = 1;
        while(true){
            if(n == Math.pow(2,k)){
                break;
            }
            k += 1;
        }
        return game(k,(double) a,(double) b,1);

    }


    public static void main(String[] args) throws Exception {
        Solution sol = new Solution();
        System.out.println(sol.solution(8,4,7));
    }
}

File: 기능구현/program_예상대진표.py
import math

def game(k,a,b,depth):
    
    if depth == k:
        return depth
    
    if a%2 == 0 and a-b == 1:
        return depth
    elif a%2 == 1 and b-a == 1:
        return depth
    else:
        return game(k,math.ceil(a/2),math.ceil(b/2),depth+1)
        

def solution(n,a,b):
    k=1
    while True:
        if n == 2**k:
            break
        k += 1

    return game(k,a,b,1)

if __name__ == "__main__":
  print(solution(8,4,7))